### PR TITLE
Add zip_open_stream()  func.

### PR DESCRIPTION
**Pull request by jinfeihan57**
@kuba-- 
I implemented zip_open_stream(), which supports decompression from memory to memory.
WDYT？

### CODE REVIEW

File: src/zip.c [hunk 1]
```diff
@@ -1022,3 +1022,19 @@ int zip_extract_stream(const char *stream, size_t size, const char *dir,
 
   return status;
 }
+
+struct zip_t *zip_open_stream(char *stream, size_t size) {
```
Comment (kuba--) [inline src/zip.c:1026]: I think it can be `const char *`

File: src/zip.c [hunk 2]
```diff
@@ -1022,3 +1022,19 @@ int zip_extract_stream(const char *stream, size_t size, const char *dir,
...
+  struct zip_t *zip = NULL;
+  zip = (struct zip_t *)calloc((size_t)1, sizeof(struct zip_t));
+  if (!zip) {
+    goto cleanup;
```
Comment (kuba--) [inline src/zip.c:1030]: if `zip` is `NULL` then there is nothing to cleanup, so we can return.

**Review by kuba-- — COMMENTED** [review]
Thanks, generally LGTM - I've put 2 minor comments.

File: src/zip.c (hunk 2 above)
Comment (jinfeihan57) [reply src/zip.c]: OK

File: src/zip.c (hunk 1 above)
Comment (jinfeihan57) [reply src/zip.c]: OK

**Comment by jinfeihan57** [comment]
@kuba-- 
I've completed the code according to your opinion. You can check it again.

**Review by kuba-- — APPROVED** [review]

**Pull request merged**